govcd/openapi_org_network.go

```diff
@@ -0,0 +1,270 @@
+	minimumApiVersion, err := orgVdcNet.client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return err
+	}
+
+	if orgVdcNet.OpenApiOrgVdcNetwork.ID == "" {
+		return fmt.Errorf("cannot delete Org Vdc network without id")
+	}
+
+	urlRef, err := orgVdcNet.client.OpenApiBuildEndpoint(endpoint, orgVdcNet.OpenApiOrgVdcNetwork.ID)
+	if err != nil {
+		return err
+	}
+
+	err = orgVdcNet.client.OpenApiDeleteItem(minimumApiVersion, urlRef, nil)
+
+	if err != nil {
+		return fmt.Errorf("error deleting Org Vdc: %s", err)
+	}
+
+	return nil
+}
+
+// GetType returns type of Org VDC network
+func (orgVdcNet *OpenApiOrgVdcNetwork) GetType() string {
+	return orgVdcNet.OpenApiOrgVdcNetwork.NetworkType
+}
+
+// IsIsolated returns true if the network type is isolated (NSX-V and NSX-T)
+func (orgVdcNet *OpenApiOrgVdcNetwork) IsIsolated() bool {
+	return orgVdcNet.GetType() == types.OrgVdcNetworkTypeIsolated
+}
+
+// IsRouted returns true if the network type is isolated (NSX-V and NSX-T)
+func (orgVdcNet *OpenApiOrgVdcNetwork) IsRouted() bool {
+	return orgVdcNet.GetType() == types.OrgVdcNetworkTypeRouted
+}
+
+// IsImported returns true if the network type is imported (NSX-T only)
+func (orgVdcNet *OpenApiOrgVdcNetwork) IsImported() bool {
+	return orgVdcNet.GetType() == types.OrgVdcNetworkTypeOpaque
+}
+
+// IsDirect returns true if the network type is direct (NSX-V only)
+func (orgVdcNet *OpenApiOrgVdcNetwork) IsDirect() bool {
+	return orgVdcNet.GetType() == types.OrgVdcNetworkTypeDirect
+}
+
+// getOpenApiOrgVdcNetworkById is a private parent for wrapped functions:
+// func (org *Org) GetOpenApiOrgVdcNetworkById(id string) (*OpenApiOrgVdcNetwork, error)
+// func (vdc *Vdc) GetOpenApiOrgVdcNetworkById(id string) (*OpenApiOrgVdcNetwork, error)
+func getOpenApiOrgVdcNetworkById(client *Client, id string, queryParameters url.Values) (*OpenApiOrgVdcNetwork, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointOrgVdcNetworks
+	minimumApiVersion, err := client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if id == "" {
+		return nil, fmt.Errorf("empty Org VDC network id")
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint, id)
+	if err != nil {
+		return nil, err
+	}
+
+	egw := &OpenApiOrgVdcNetwork{
+		OpenApiOrgVdcNetwork: &types.OpenApiOrgVdcNetwork{},
+		client:               client,
+	}
+
+	err = client.OpenApiGetItem(minimumApiVersion, urlRef, queryParameters, egw.OpenApiOrgVdcNetwork)
+	if err != nil {
+		return nil, err
+	}
+
+	return egw, nil
+}
+
+// returnSingleOpenApiOrgVdcNetwork helps to reduce code duplication for `GetOpenApiOrgVdcNetworkByName` functions with different
+// receivers
+func returnSingleOpenApiOrgVdcNetwork(name string, allEdges []*OpenApiOrgVdcNetwork) (*OpenApiOrgVdcNetwork, error) {
+	if len(allEdges) > 1 {
+		return nil, fmt.Errorf("got more than 1 Org Vdc network by name '%s' %d", name, len(allEdges))
```

Comment:
Suggesting to keep user messages human-oriented:
```suggestion
		return nil, fmt.Errorf("got more than 1 Org VDC network by name '%s' %d", name, len(allEdges))
```
```suggestion
		return nil, fmt.Errorf("got more than one Org Vdc network by name '%s' %d", name, len(allEdges))
```

Reply:
Changed